carddav/server.go

```diff
@@ -262,7 +262,12 @@ func (b *backend) HeadGet(w http.ResponseWriter, r *http.Request) error {
 	}
 
 	w.Header().Set("Content-Type", vcard.MIMEType)
-	// TODO: set ETag, Last-Modified
+	if ao.ETag != "" {
+		w.Header().Set("ETag", ao.ETag)
```

Comment:
The ETag needs to be quoted here. The marshalling can be done via: `internal.ETag(ao.ETag).String()`.

See `backend.HeadGet` in `/server.go` for a reference.